src/framework/musesampler/internal/musesamplerconfiguration.cpp

```diff
@@ -23,27 +23,48 @@
 #include "musesamplerconfiguration.h"
 
 #include <cstdlib>
+#include <QStandardPaths>
 
 using namespace mu;
 using namespace mu::musesampler;
 
 #if defined(Q_OS_LINUX)
-static const io::path_t DEFAULT_PATH("libMuseSamplerCoreLib.so");
+static const io::path_t DEFAULT_LIB_NAME("libMuseSamplerCoreLib.so");
+static io::path_t DEFAULT_PATH()
+{
+    return QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/MuseSampler/lib/" + DEFAULT_LIB_NAME;
+}
 #elif defined(Q_OS_MAC)
-static const io::path_t DEFAULT_PATH("/usr/local/lib/libMuseSamplerCoreLib.dylib");
+static const io::path_t DEFAULT_LIB_NAME("libMuseSamplerCoreLib.dylib");
```

Comment:
This used to be specified as a full path on macOS for the system directory; this worked as a fallback without this for me, though, on Monterey.  Any reason the full absolute path was used here before?